docusaurus/docs/how-to-guides/add-user-storage.md

```diff
@@ -0,0 +1,80 @@
+---
+id: add-user-storage
+title: Add user storage to your plugin
+description: How to add individual user storage to your plugin.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - storage
+---
+
+# Add user storage to your plugin
+
+Add user storage to your plugin to store user-specific data in the Grafana backend. This data is stored per user and is accessible only to that user. It will also persist independently of the user browser. Note that this data is not encrypted and should not be used to store sensitive information. The typical use case for user storage is to store user preferences or settings.
+
+:::important
+
+This feature is only available in Grafana 11.5 and later.
+
+:::
+
+## Sample query editor
+
+Let's say you have a `QueryEditor` that looks similar to the example below. It has a `Select` field where you can select the kind of query result that you expect to return. We will add user storage to this component to save the user's preferred query type so that it can be used as the default next time the user opens the query editor.
+
+```tsx
+import React, { ReactElement, useEffect } from 'react';
+import { InlineFieldRow, InlineField, Select } from '@grafana/ui';
+import { QueryEditorProps, SelectableValue } from '@grafana/data';
+import { DataSource } from 'datasource';
+import { MyDataSourceOptions, MyQuery } from 'types';
+import { usePluginUserStorage } from '@grafana/runtime';
+
+type Props = QueryEditorProps<DataSource, MyQuery, MyDataSourceOptions>;
+
+export function QueryEditor(props: Props): ReactElement {
+  const { query, onChange } = props;
+  const queryTypes = [
+    {
+      label: 'Timeseries',
+      value: 'timeseries',
+    },
+    {
+      label: 'Table',
+      value: 'table',
+    },
+  ];
+  const storage = usePluginUserStorage();
```

Comment:
This will result in a broken plugin in case the runtime is <11.5.0. I think all examples where new Grafana features are being used in a plugin should be backwards compatible. Can you add a runtime check to see if the hook is exposed by the runtime package?

Reply:
That would clutter the doc quite a bit. I agree that may be the best for a complete example in the examples repo but here, for this summarized example, I'd rather keep it as simple as possible (I also don't see we are doing this with other docs examples like this [usePluginLinks](https://github.com/grafana/plugin-tools/blob/userStorage/docusaurus/docs/how-to-guides/ui-extensions/create-an-extension-point.md#creating-an-extension-point-for-links)).

We are already saying that the minimum version for this is 11.5 so hope that's enough. It would be nice to have a generic section / doc for "how to use new features maintaining Grafana version compatibility" that we could link everytime we document a new feature so we don't clutter all the examples 🤔

Reply:
No that's fair enough. I'm planning to add an article on this topic. Let's leave it like this for now 👍